cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java

```diff
@@ -545,424 +550,487 @@ public Node getThisNode() {
+            }
+          }
 
-  /**
-   * Forward a plan to a node using the default client.
-   *
-   * @param plan
-   * @param node
-   * @param header must be set for data group communication, set to null for meta group
-   *               communication
-   * @return a TSStatus indicating if the forwarding is successful.
-   */
-  TSStatus forwardPlan(PhysicalPlan plan, Node node, Node header) {
-    if (node == thisNode || node == null) {
-      logger.debug("{}: plan {} has no where to be forwarded", name, plan);
-      return StatusUtils.NO_LEADER;
-    }
+          if (allLogsValid) {
+            if (logger.isDebugEnabled()) {
+              logger.debug("{} makes {} catch up with {} cached logs", name, follower, logs.size());
+            }
+            catchUpService.submit(new LogCatchUpTask(logs, follower, this));
+          } else {
+            logger.debug("{}: Logs in {} are too old, catch up with snapshot", name, follower);
+            catchUpService.submit(new SnapshotCatchUpTask(logs, snapshot, follower, this));
+          }
+        } else {
+          lastCatchUpResponseTime.remove(follower);
+          logger.warn("{}: Catch-up failed: node {} is currently unavailable", name, follower);
+        }
+      }
 
-    logger.info("{}: Forward {} to node {}", name, plan, node);
+      public String getName () {
+        return name;
+      }
 
-    AsyncClient client = connectNode(node);
-    if (client != null) {
-      return forwardPlan(plan, client, node, header);
-    }
-    return StatusUtils.TIME_OUT;
-  }
+      /**
+       * @return the header of the data raft group or null if this is in a meta group.
+       */
+      public Node getHeader () {
+        return null;
+      }
 
-  /**
-   * Forward a non-query plan to "receiver" using "client".
-   * @param plan a non-query plan
-   * @param client
-   * @param receiver
-   * @param header to determine which DataGroupMember of "receiver" will process the request.
-   * @return a TSStatus indicating if the forwarding is successful.
-   */
-  TSStatus forwardPlan(PhysicalPlan plan, AsyncClient client, Node receiver, Node header) {
-    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-    DataOutputStream dataOutputStream = new DataOutputStream(byteArrayOutputStream);
-    try {
-      plan.serializeTo(dataOutputStream);
-      AtomicReference<TSStatus> status = new AtomicReference<>();
-      ExecutNonQueryReq req = new ExecutNonQueryReq();
-      req.setPlanBytes(byteArrayOutputStream.toByteArray());
-      if (header != null) {
-        req.setHeader(header);
+      /**
+       <<<<<<< HEAD
+       * Forward a plan to a node using the default client.
+       *
+       * @param plan
+       * @param node
+      =======
+       * Forward a non-query plan to a node using the default client.
+       * @param plan a non-query plan
+       * @param node cannot be the local node
+      >>>>>>> 378da6117d05a7e38d05fb46cb20b3023434c08c
```

Comment:
Please notice the conflict marks and resolve them. I will mark all I see in case you miss any.

Reply:
Thanks for your reminding. I will remove them.